app/src/vuex/modules/session.js

```diff
@@ -136,20 +156,59 @@ export default ({ apollo }) => {
     },
     async signIn(
       { state, getters: { currentNetwork }, commit, dispatch },
-      { address, sessionType = `ledger`, networkId }
+      { address, sessionType = `ledger`, HDPath, curve, networkId }
     ) {
+      if (!HDPath || !curve) {
+        // first search in localStorage
+        const session = JSON.parse(
+          localStorage.getItem(`cosmos-wallets-${address}`)
+        )
+        if (
+          session &&
+          session.HDPath &&
+          session.curve &&
+          address === session.address
+        ) {
+          HDPath = session.HDPath
+          curve = session.curve
+        } else {
+          // set default if this is not defined
+          HDPath = currentNetwork.defaultHDPath
+          curve = currentNetwork.defaultCurve
+          // store in localStorage for later use
+          if (session)
+            localStorage.setItem(
+              `cosmos-wallets-${address}`,
+              JSON.stringify({
+                ...session,
+                HDPath,
+                curve,
+              })
+            )
+        }
+      }
```

Comment:
Maybe I can still add the tests for this bit? Is it worth it?

Reply:
Doesn't look too important. According to codecov what is left to test is just an assignment, nothing critical:

![image](https://user-images.githubusercontent.com/40721795/87601136-bf2def80-c6f4-11ea-8048-ec0cc411fd0e.png)

Reply:
ping